### PR TITLE
Check GitHub account

### PR DESCRIPTION
**Pull request by tropicar**
Check if there is a git account set up in order to avoid the bot account overrides the user git account data.

### CODE REVIEW

File: src/commands/githubActions/bumpUpstream/index.ts [hunk 1]
```diff
@@ -158,8 +160,11 @@ Compose - ${JSON.stringify(compose, null, 2)}
     return;
   }
 
-  await shell(`git config user.name ${userName}`);
-  await shell(`git config user.email ${userEmail}`);
+  //Check if there is a github account set up, in other case it uses the default account
+  if (!(isGitUserSet() && isGitEmailSet())) {
+    await shell(`git config user.name ${userName}`);
+    await shell(`git config user.email ${userEmail}`);
+  }
```
Comment (dapplion) [inline src/commands/githubActions/bumpUpstream/index.ts:167]:
```ts
if (!(isGitUserSet())) {
    await shell(`git config user.name ${userName}`);
}
if (!(isGitEmailSet())) {
    await shell(`git config user.email ${userEmail}`);
}
```

File: src/utils/isGitUserSet.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    // we need to check it out if thee value exists
+    if (currentUser.length > 0) {
+      return true;
+    } else {
+      return false;
+    }
+  } catch (e) {
+    // Empty value or not value defined returns shell error, we catch that error and return false
+    console.log("here");
```
Comment (dapplion) [inline src/utils/isGitUserSet.ts:15]: Remove console.log

File: src/utils/isGitUserSet.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export async function isGitUserSet(): Promise<boolean> {
+  try {
+    const currentUser = await shell(`git config user.name`);
+
+    // we need to check it out if thee value exists
+    if (currentUser.length > 0) {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (dapplion) [inline src/utils/isGitUserSet.ts:12]:
```ts
currentUser.length > 0
```

**Review by dapplion — COMMENTED** [review]
Can you fix CI failures?

File: src/commands/githubActions/bumpUpstream/index.ts [hunk 4]
```diff
@@ -158,8 +160,14 @@ Compose - ${JSON.stringify(compose, null, 2)}
     return;
   }
 
-  await shell(`git config user.name ${userName}`);
-  await shell(`git config user.email ${userEmail}`);
+  //Check if there is a github account set up, in other case it uses the default account
+
+  if (!isGitUserSet()) {
```
Comment (dapplion) [inline src/commands/githubActions/bumpUpstream/index.ts:165]: You must await this statement or it will always be true. Same for below

**Comment by pablomendezroyo** [comment]
Requires more work

**Pull request closed without merging**